cpp/include/cudf/utilities/traits.hpp

```diff
@@ -282,4 +282,36 @@ constexpr inline bool is_simple(data_type type) {
   return not is_compound(type);
 }
 
+/**---------------------------------------------------------------------------*
+ * @brief Indicates whether the type `T` is an arithmetic type.
+ *
+ * @tparam T  The type to verify
+ * @return true `T` is arithmetic
+ * @return false `T` is not arithmetic
+ *---------------------------------------------------------------------------**/
+template <typename T>
+constexpr inline bool is_arithmetic() {
+  return std::is_integral<T>::value or
+         std::is_floating_point<T>::value or
+         cudf::is_timestamp<T>();
```

Comment:
does timestamp support all arithmetic operations?

Reply:
Definitely not. This should just use `std::is_arithmetic<T>`.

Reply:
How about `cudf::is_numeric` ? that seems to be identical to`std::is_arithmetic`. I'm not sure which is more appropriate, with regards to context/intent.